pkg/services/accesscontrol/filter.go

```diff
@@ -3,101 +3,65 @@ package accesscontrol
 import (
 	"context"
 	"errors"
-	"fmt"
+	"strconv"
 	"strings"
 
 	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
 )
 
 var sqlIDAcceptList = map[string]struct{}{
 	"org_user.user_id": {},
 }
 
-type SQLDialect interface {
-	DriverName() string
-}
+const denyQuery = " 1 = 0"
+const allowAllQuery = " 1 = 1"
 
 // Filter creates a where clause to restrict the view of a query based on a users permissions
 // Scopes for a certain action will be compared against prefix:id:sqlID where prefix is the scope prefix and sqlID
 // is the id to generate scope from e.g. user.id
-func Filter(ctx context.Context, dialect SQLDialect, sqlID, prefix, action string, user *models.SignedInUser) (string, []interface{}, error) {
+func Filter(ctx context.Context, sqlID, prefix, action string, user *models.SignedInUser) (string, []interface{}, error) {
 	if _, ok := sqlIDAcceptList[sqlID]; !ok {
-		return "", nil, errors.New("sqlID is not in the accept list")
+		return denyQuery, nil, errors.New("sqlID is not in the accept list")
 	}
-
 	if user.Permissions == nil || user.Permissions[user.OrgId] == nil {
-		return "", nil, errors.New("missing permissions")
+		return denyQuery, nil, errors.New("missing permissions")
 	}
 
-	scopes := user.Permissions[user.OrgId][action]
-	if len(scopes) == 0 {
-		return " 1 = 0", nil, nil
+	var hasWildcard bool
+	var ids []interface{}
+	for _, scope := range user.Permissions[user.OrgId][action] {
+		if strings.HasPrefix(scope, prefix) {
+			if scope[len(scope)-1] == '*' {
```

Comment:
Could we, instead, trim the prefix and compare to the string `"*"`? 🤔 I can't say for sure that this is problematic, but I'm a little bit worried about how this would behave given say (silly example) `dashboards:id:54*`

Reply:
I guess we could. But we will have to try both  strings.TrimPrefix(scope, prefix + ":") and strings.TrimPrefix(scope, prefix + ":id:")